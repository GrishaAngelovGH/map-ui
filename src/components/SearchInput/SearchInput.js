import { useState } from 'react'
import FormControl from 'react-bootstrap/FormControl'

const SearchInput = ({ placeholder, onChange }) => {
  const [value, setValue] = useState('')

  const handleInputChange = ({ target }) => {
    setValue(target.value)
    onChange(target.value)
  }

  const handleClearInput = () => {
    setValue('')
    onChange('')
  }

  return (
    <div className='d-flex position-relative'>
      <i className='bi bi-search text-secondary fs-4 ps-2 position-absolute'></i>
      <FormControl
        value={value}
        placeholder={placeholder}
        className='mb-3 ps-5'
        onChange={handleInputChange}
      />
      <i role='button' className='bi bi-backspace text-secondary fs-4 ps-2' onClick={handleClearInput}></i>
    </div>
  )
}

export default SearchInput