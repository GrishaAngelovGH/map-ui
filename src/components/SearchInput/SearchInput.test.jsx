import { render } from '@testing-library/react'

import SearchInput from './SearchInput'

test('should render SearchInput component', () => {
  const view = render(<SearchInput placeholder='Search' />)

  expect(view).toMatchSnapshot()
})