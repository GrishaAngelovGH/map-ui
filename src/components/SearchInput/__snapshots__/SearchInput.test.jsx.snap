// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render SearchInput component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex position-relative"
      >
        <i
          class="bi bi-search text-secondary fs-4 ps-2 position-absolute"
        />
        <input
          class="mb-3 ps-5 form-control"
          placeholder="Search"
          value=""
        />
        <i
          class="bi bi-backspace text-secondary fs-4 ps-2"
          role="button"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex position-relative"
    >
      <i
        class="bi bi-search text-secondary fs-4 ps-2 position-absolute"
      />
      <input
        class="mb-3 ps-5 form-control"
        placeholder="Search"
        value=""
      />
      <i
        class="bi bi-backspace text-secondary fs-4 ps-2"
        role="button"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
